eng/BootStrapMSBuild.targets

```diff
@@ -190,6 +190,11 @@
     <Copy SourceFiles="@(InstalledSdks)"
           DestinationFiles="@(InstalledSdks -> '$(BootstrapDestination)Sdks\%(RecursiveDir)%(Filename)%(Extension)')" />
 
+    <!-- The .NET SDK has a dependency on DependencyModel, but relies on having it "next to MSBuild" in the final
+         layout. Copy it from there to our little weirdo bootstrap layout next to the SDK tasks. -->
+    <Copy SourceFiles="$(DOTNET_INSTALL_DIR)\sdk\$(DotNetCliVersion)\Microsoft.Extensions.DependencyModel.dll"
+          DestinationFolder="$(BootstrapDestination)Sdks\Microsoft.NET.Sdk\tools\net6.0" />
```

Comment:
Is this right next to the MSBuild binary? I would've guessed $(BootstrapDestination)$(TargetMSBuildToolsVersion)\Bin

If not, why is this place better?

Reply:
> I would've guessed $(BootstrapDestination)$(TargetMSBuildToolsVersion)\Bin

BootstrapDestination already resolves to `<reporoot>\artifacts\bin\bootstrap\<TARGETFRAMEWORK>\MSBuild` (the folder where the binaries exist already).

> If not, why is this place better?

Looks like he tried it based on e319909, but I'm also curious

Reply:
Thanks for calling me on this, it caused me to catch a bad bug. The short answer is "because of #6558". Longer answer and a question there.

Reply:
This approach will work even if we fix that, so I'm going to stick with it but update the comment.